Use SimpleBar scroll bar in 'Create Stream' UI

raghavluthra20 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes: #21307 

**Testing plan:** <!-- How have you tested? -->
manual

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
Before:

https://user-images.githubusercontent.com/59819879/156878901-084e3363-2847-4a4d-975c-f1b00d35355c.mp4

After:

https://user-images.githubusercontent.com/59819879/157176123-d936e9c1-b1e9-4529-b6e8-d474f53dd876.mp4








<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

raghavluthra20 commented on static/styles/subscriptions.css:
```diff
@@ -617,6 +617,10 @@ h4.stream_setting_subsection_title {
 }
 
+#create-stream-simplebar-wrapper {
+    max-height: calc(95vh - 145px);
```
This value was found by trial and error. A similar approach was used in other places within this file as well.
(for example: line 224)

raghavluthra20 commented on static/styles/subscriptions.css:
```diff
@@ -786,7 +790,7 @@ h4.stream_setting_subsection_title {
     #stream-creation {
         max-height: calc(100% - 102px);
-        overflow: auto;
+        overflow: hidden;
```
This rule had to be changed to `hidden` to hide the regular scroll bar.

alya commented:
@juliaBichler01 would you be up for reviewing this one? Thanks!

juliaBichler01 commented on static/templates/stream_settings/stream_creation_form.hbs:
```diff
@@ -3,36 +3,38 @@
-                <div id="stream_subscription_error" class="stream_creation_error"></div>
-                <div class="controls" id="people_to_add"></div>
-            </section>
+        <div id="create-stream-simplebar-wrapper">
```
Same issue as with `stream_edit.js`, why did you create this manually?

juliaBichler01 commented:
@alya sure, happy to!

juliaBichler01 reviewed (changes requested):
I added a few comments, but apart from that it looks promising!

raghavluthra20 commented on static/templates/stream_settings/stream_creation_form.hbs:
```diff
@@ -3,36 +3,38 @@
+                </section>
+                <section class="block">
+                    <label class="stream-title" for="people_to_add">
+                        {{t "People to add" }}
```
This was added in #21233. This must've been added when I was resolving conflicts for this file. Perhaps a rebase and push will fix this.

raghavluthra20 commented:
Hey @juliaBichler01. I think I have addressed all the comments apart from [this](https://github.com/zulip/zulip/pull/21313#discussion_r821812455) last one. Can you advice me on how to go about adding a handler for this in `stream_create.js`? I'm going through the code in `stream_create.js` right now looking for something that might give me an idea on how to accomplish this.

timabbott commented:
@sahil839 it might be helpful for you to do a review on this as well at this point.

sahil839 commented:
I played with the PR and I can see that the error is always fixed at the top and not in the scrollable container and thus I cannot scroll till bottom (some subscribers are not visibile in the shared GIF). Do we want such behavior? Sharing the GIF -
![stream-simplebar](https://user-images.githubusercontent.com/35494118/160135585-0cf9eda6-9d99-490c-9725-791f75be8653.gif)

And maybe the correct soluttion for the narrow windows behavior is to change the height that you have set for that case since the `Create stream` title at top is removed.

sahil839 commented on static/templates/stream_settings/stream_creation_form.hbs:
```diff
@@ -3,36 +3,38 @@
-                <div id="stream_subscription_error" class="stream_creation_error"></div>
-                <div class="controls" id="people_to_add"></div>
-            </section>
+        <div id="create-stream-simplebar-wrapper" data-simplebar>
```
I think we can avoid adding this extra element and instead add `data-simplebar` to `stream-creation-body` element itself.

raghavluthra20 replied:
The problem with doing that was the scrollbar was overflowing when the number of users increased; So I decided to use an extra div which eliminated that problem. Sharing the videos below:

https://user-images.githubusercontent.com/59819879/160231538-e7d0bf85-512f-4af9-87ab-694bc469930e.mp4


https://user-images.githubusercontent.com/59819879/160231548-56a04434-afe9-458c-bbf8-58ef6fa6768d.mp4

sahil839 commented:
Without using extra div, we can set the height something different, soemthing around `calc(95vh - 160px)` (I found this by trial and error method) and then it works fine. Though, it is not ideal to set these exact value of heights by trial and error method.

raghavluthra20 commented:
> Without using extra div, we can set the height something different, soemthing around `calc(95vh - 160px)` (I found this by trial and error method) and then it works fine. Though, it is not ideal to set these exact value of heights by trial and error method.

Ah, I see. I was using the `max-height` rule, but it works fine with just the `height` rule. Thanks for the help.
Also, a question to tackle problems like this in the future: What is a good way of determining values like these if not by trial and error?

alya commented:
Hm, I just tested this out, and it's not quite right. The bottom of the text doesn't quite fit, but there's no scroll bar until I make the window smaller than this:
![Screen Shot 2022-04-06 at 10 51 28 PM](https://user-images.githubusercontent.com/2090066/162129600-acb0dd41-71da-4fbb-9b65-ea7a09ac87e6.png)

raghavluthra20 commented:
> Hm, I just tested this out, and it's not quite right. The bottom of the text doesn't quite fit, but there's no scroll bar until I make the window smaller than this: ![Screen Shot 2022-04-06 at 10 51 28 PM](https://user-images.githubusercontent.com/2090066/162129600-acb0dd41-71da-4fbb-9b65-ea7a09ac87e6.png)

@alya Thanks for pointing it out. I'm looking into it.

alya commented:
@raghavluthra20 will you be able to continue iterating on this PR, or should someone else take over from here?

raghavluthra20 commented:
> @raghavluthra20 will you be able to continue iterating on this PR, or should someone else take over from here?

I looked at it again and I seem to be stuck... I guess someone else can take this issue up.

alya commented:
OK, no worries -- thanks for giving it a try!

alya commented:
@sahil839 or @juliaBichler01 , perhaps one of you could take over this one?

juliaBichler01 commented:
@alya i can definitively try this, it should be relatively similar to my other PR :)

timabbott commented:
Closing in favor of #22943, thanks @raghavluthra20!
